h/api/test/uri_test.py

```diff
@@ -154,6 +153,13 @@
     ("http://example.com?gclid_foo=abcde", "http://example.com?gclid_foo=abcde"),
     ("http://example.com?bar_gclid=abcde", "http://example.com?bar_gclid=abcde"),
     ("http://example.com?WT=abcde", "http://example.com?WT=abcde"),
-])
+]
+
+
+@pytest.mark.parametrize("url_in,url_out", TEST_URLS)
 def test_normalize(url_in, url_out):
     assert uri.normalize(url_in) == url_out
+
+@pytest.mark.parametrize("url,_", TEST_URLS)
```

Comment:
The test file already imports `unicode_literals`.

Since the implementation has 3 different return statements, and each of those converts the value first to `text_type` before returning, I thought it is going to be safer to run all the test cases through the `isinstance` assertion. This will also ensure that any changes in the future (maybe another if statement with an early return) will be tested automatically.

Thoughts?

I will change the assertion to use `text_type` rather than `unicode` though, that's a good poing.